msteams-platform/bots/how-to/conversations/Workflow bot in Teams.md

```diff
@@ -0,0 +1,460 @@
+
+You can update the initialization logic to:
+
+1. Set `options.adapter` to use your own `BotFrameworkAdapter`.
+1. Set `options.command.commands` to include more command handlers.
+1. Set `options.cardAction.actions` to include more action handlers.
+1. Set `options.{feature}.enabled` to enable more `ConversationBot` functionality.
+
+For more information on initialization customization, see [additional initialization customization](https://github.com/OfficeDev/TeamsFx/wiki/Respond-to-chat-commands-in-Teams#customize-initialization)
+
+## Add card action
+
+Steps you can follow to add card actions:
+
+1. [Add an action to your Adaptive Card](#add-an-action-to-your-adaptive-card)
+1. [Add Adaptive Card for action response](#add-adaptive-card-for-action-response)
+1. [Add action handler](#add-action-handler)
+1. [Register the action handler](#register-the-action-handler)
+
+### Add an action to your Adaptive Card
+
+User universal action `Action.Execute` defines your action in an Adaptive Card, which can be rendered as a button in the card. `Action.Execute` is a sample action:
+
+```
+
+{ 
+  "type": "AdaptiveCard", 
+  "body": [
+    ...
+    {
+      "type": "ActionSet",
+      "actions": [
+        {
+          "type": "Action.Execute",
+          "title": "DoStuff",
+          "verb": "doStuff" 
+        }
+      ]
+    }
+  ]
+  ... 
+}
+```
+
+> [!NOTE]
+> When the action is invoked in Teams, verb property is required, so that the TeamsFx conversation SDK can invoke the corresponding action handler. Ensure to provide a global unique string for the verb property, when you're using a general string that might cause a collision with other bot. This can avoid an unexpected behavior.
+
+### Add Adaptive Card for action response
+
+You can return a new Adaptive Card for each action invoked to display the response to end user. You can use [Adaptive Card designer](https://adaptivecards.io/designer/) to design your card layout according to your business need.
+
+To get started, you can create a sample card (`responseCard.json`) with the following content, and put it in `bot/src/adaptiveCards` folder:
+
+```responseCard.json
+{
+  "type": "AdaptiveCard",
+  "body": [
+    {
+      "type": "TextBlock",
+      "size": "Medium",
+      "weight": "Bolder",
+      "text": "This is a sample action response."
+    }
+  ],
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "version": "1.4"
+}
+```
+
+### Add action handler
+
+Add action handler to implement `TeamsFxAdaptiveCardActionHandler` for processing the logic when corresponding action is executed.
+
+* `triggerVerb` is the verb property of your action.
+* `actionData` is the data associated with the action, which may include dynamic user input or some contextual data provided in the data property of your action.
+* If an Adaptive Card is returned, the existing card is replaced with it by default.
+
+    ```
```

Comment:
Can we provide two tabs for the js/ts code snippet in this docs, like below:
![image](https://user-images.githubusercontent.com/10163840/198512726-14c98ad5-6c40-4d6c-8345-ba2498eefe24.png)

Reply:
yes, this can be done.